v2/protocol/pubsub/message.go

```diff
@@ -0,0 +1,123 @@
+package pubsub
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/binding/format"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
+)
+
+const (
+	prefix      = "ce-"
+	contentType = "Content-Type"
+)
+
+var specs = spec.WithPrefix(prefix)
+
+// Message represents a Pub/Sub message.
+// This message *can* be read several times safely
+type Message struct {
+	Msg pubsub.Message
+
+	format   format.Format
+	version  spec.Version
+	encoding binding.Encoding
+}
+
+// NewMessage returns a binding.Message with data and attributes.
+// This message *can* be read several times safely
+func NewMessage(data []byte, attributes map[string]string) *Message {
+	encoding := binding.EncodingBinary
```

Comment:
What if the message is a non-cloudevent? In that case, this function should return `binding.EncodingUnknown`

Reply:
removed encoding as field and used same strategy as https://github.com/cloudevents/sdk-go/blob/master/v2/protocol/kafka_sarama/message.go#L78, iiuc, this should now be done properly.